### DISCUSSION TITLE
Web Framework support python

### DISCUSSION THREAD

**Discussion by c-14795** [pinpoint-apm/pinpoint-c-agent#441] [Q&A] [answered]
Does pinpoint python support fast api?

**Comment by EyelynSu** [comment, marked as answer]
Hello~
Welcome to user pinpoint!
`fast api` -> Do you mean the Framework [FastAPI](https://fastapi.tiangolo.com) ? 
We have only released the middleware for FastAPI till now. 
FastAPI middleware, please ref to https://github.com/pinpoint-apm/pinpoint-c-agent/tree/master/plugins/PY/pinpointPy/Fastapi

How to use:
```
from pinpointPy import monkey_patch_for_pinpoint, set_agent
from starlette.middleware import Middleware
from pinpointPy.Fastapi.middleware import PinPointMiddleWare

middleware = [
    Middleware(PinPointMiddleWare)
]
app = FastAPI(middleware=middleware)
server = Server()

# enable auto-interceptor plugins
monkey_patch_for_pinpoint()

# set pinpoint related environment
# signature: set_agent(app_id_str: str, app_name_str: str, collect_agent_host: str, trace_limit: int = -1, debug_callback=True)
set_agent("app id", "app name", 'collector-agent's address')

```
By the way, if you want to monitor other components, such as mysql, redis, etc. please give us a list of  which library you are using.
We will try to support them.

Look forward to your reply.

**Reply by c-14795** [reply]
Hello, Thanks for the response, it would be great if support for redis is released. 

Also from  which version of fastapi this is supported ?

**Reply by EyelynSu** [reply]
By the way, which lib and version do you use for redis? [aioredis](https://pypi.org/project/aioredis/) , [redis](https://pypi.org/project/redis/), or any other one?

**Reply by c-14795** [reply]
we use aioredis==1.3.1, from which version of fast api pinpoint is supported ?

**Reply by EyelynSu** [reply]
fastapi==0.75.2 is OK, other version haven't been tested. What version you are using?

**Reply by c-14795** [reply]
fastapi==0.63.0

**Reply by EyelynSu** [reply]
@c-14795 Hello~
We have supported fastapi+redis, and we released it on https://test.pypi.org. Please use pip install -i https://test.pypi.org/simple/ pinpointPy to have a try.
How to use:
```
# pinpoint
##############################################
from starlette_context.middleware import ContextMiddleware
from starlette_context import context, plugins

from pinpointPy import set_agent
from pinpointPy.Fastapi import asyn_monkey_patch_for_pinpoint
from starlette.middleware import Middleware
from pinpointPy.Fastapi import PinPointMiddleWare

middleware = [
    Middleware(ContextMiddleware),
    Middleware(PinPointMiddleWare)
]
asyn_monkey_patch_for_pinpoint()
set_agent("fastapi-redis", "fastapi-redis", 'tcp:collect-agent address:9999', -1, True)
##############################################
app = FastAPI(title='FastAPI Redis', middleware=middleware)
```

**Reply by eeliu** [reply]
please try the https://github.com/pinpoint-apm/pinpoint-c-agent/tree/dev/testapps/fastapi
